ultravox/tools/ds_tool/ds_tool.py

```diff
@@ -115,32 +117,46 @@ def map_split(
         exclude_fields: List[str],
     ) -> datasets.Dataset:
         print(f'Generating "{self.new_column_name}" with template:\n{self.template}')
-        return ds_split.map(
+        ds_mapped = ds_split.map(
             lambda sample: self._map_sample(sample, set(exclude_fields)),
             num_proc=num_proc,
             writer_batch_size=writer_batch_size,
         )
+        if self.template_failures == 0:
+            return ds_mapped
+
+        # Filter out samples where new_column_name is None
+        return ds_mapped.filter(
+            lambda sample: (True if sample[self.new_column_name] != None else False),
+            num_proc=num_proc,
+            writer_batch_size=writer_batch_size,
+        )
 
     def _map_sample(self, sample, exclude_fields):
         # using a Jinja template for some added flexibility, template can include variables and functions
         # e.g., {{ text }} or {{ text_proc.format_asr_text(text) }}
         try:
-            # We need to filter out the audio before the sample is passed into the jinja template
-            # or it will get loaded into memory and spike usage.
+            # Filter out the audio before the sample is passed into the jinja template, or it will get loaded into memory.
             filtered_sample = {
                 k: sample[k] for k in sample.keys() if k not in exclude_fields
             }
-
             rendered = jinja2.Template(
                 self.template, undefined=jinja2.StrictUndefined
             ).render(**filtered_sample, json_dump=json.dumps, text_proc=text_proc)
+
         except jinja2.TemplateError as e:
-            print(f"Error rendering template: {e}")
-            print(f"template: {self.template}")
-            print(f"sample keys: {list(filtered_sample.keys())}")
-            raise ValueError(
-                f"Template rendering failed. Make sure all keys in the template exist in the sample."
-            ) from e
+            error_message = str(e)
+            if "The formatted text is empty." in error_message:
```

Comment:
Glad to see this worked out, but this check seems a bit fragile. Is there no way to distinguish these two cases by the exception type?